### DISCUSSION TITLE
Hiding for Header

### DISCUSSION THREAD

**Discussion by tangjh** [awcodes/filament-table-repeater#107] [Ideas]
May I request to add the ability to hide a header based on condition?

    public function hidden(bool | Closure | null $condition = true): static
    {
        $this->isHidden = $condition;

        return $this;
    }

    public function isHidden(): bool
    {
        return $this->evaluate($this->isHidden)
            ?? false;
    }

resource file:

                        @foreach ($headers as $key => $header)
                        @if (!$header->isHidden())
                            <th
                                @class([
                                    'table-repeater-header-column p-2 font-medium first:rounded-tl-xl last:rounded-tr-xl bg-gray-100 dark:text-gray-300 dark:bg-gray-900/60',
                                    match($header->getAlignment()) {
                                      'center', Alignment::Center => 'text-center',
                                      'right', 'end', Alignment::Right, Alignment::End => 'text-end',
                                      default => 'text-start'
                                    }
                                ])
                                style="width: {{ $header->getWidth() }}"
                            >
                                {{ $header->getLabel() }}
                                @if ($header->isRequired())
                                    <span class="whitespace-nowrap">
                                        <sup class="font-medium text-danger-700 dark:text-danger-400">*</sup>
                                    </span>
                                @endif
                            </th>
                        @endif
                        @endforeach

**Comment by awcodes** [comment]
Hiding headers would break the table and make the table not accessible.

**Comment by Willis1776** [comment]
So I have columns that are visible based on the user's permissions. The header will still show unfortunately. Is there a way to make it say the header could be hidden based on the same permission. For example,
`              Header::make('markup')->width('100px')->visible(fn() => Auth::user()->can('view_pricing_invoice')),`
`              TextInput::make('markup')->prefix('%')->live(debounce: 500)->visible(fn() => Auth::user()->can('view_pricing_invoice')),`

**Comment by awcodes** [comment]
Sorry, it's not possible to do this and have it be a semantic html table. You can show or hide a column but it can't be conditional based on row values.
